exist-core/src/test/java/org/exist/xquery/AbsolutePathTests.java

```diff
@@ -0,0 +1,160 @@
+import static org.exist.test.XQueryAssertions.assertXQResultSimilar;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Ensure absolute XPath expression will raise a static error.
+ *
+ * @author <a href="mailto:juri@existsolutions.com">Juri Leino</a>
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
+ */
+public class AbsolutePathTests extends XQueryCompilationTest {
+    @Test
+    public void declaredFunctionAbsoluteSlash() throws EXistException, PermissionDeniedException {
+        final String query = "declare function local:x() { /x }; local:x()";
+        final String expectedMessage = "Leading '/' selects nothing, ContextItem is absent in function body";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1,30, expectedMessage, compileQuery(query));
+    }
+
+    @Test
+    public void declaredFunctionLoneSlash() throws EXistException, PermissionDeniedException {
+        final String query = "declare function local:x() { / }; local:x()";
+        final String expectedMessage = "Leading '/' selects nothing, ContextItem is absent in function body";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1, 30, expectedMessage, compileQuery(query));
+    }
+
+    @Test
+    public void declaredFunctionAbsoluteDoubleSlash() throws EXistException, PermissionDeniedException {
+        final String query = "declare function local:x() { //x }; local:x()";
+        final String expectedMessage = "Leading '//' selects nothing, ContextItem is absent in function body";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1, 30, expectedMessage, compileQuery(query));
+    }
+
+    @Test
+    public void immediateLambdaContainsSlash() throws EXistException, PermissionDeniedException {
+        final String query = "(function() { /x })()";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1, 15, compileQuery(query));
+    }
+
+    @Test
+    public void immediateLambdaContainsLoneSlash() throws EXistException, PermissionDeniedException {
+        final String query = "(function() { / })()";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1, 15, compileQuery(query));
+    }
+
+    @Test
+    public void immediateLambdaContainsDoubleSlash() throws EXistException, PermissionDeniedException {
+        final String query = "(function() { //x })()";
+        assertXQStaticError(ErrorCodes.XPDY0002, 1, 15, compileQuery(query));
+    }
+
+    @Test
+    public void immediateLambdaWithDocumentAndDoubleSlash() throws EXistException, PermissionDeniedException {
+        final Source expected = elemSource("<result><x/><x/><x/></result>");
+
+        final String query = "let $f := function($d) { $d ! //x }\n" +
+                "let $d := document { <root><x/><x/><y><x/></y></root> }\n" +
+                "return\n" +
+                "  <result>{ $f($d) }</result>";
+        final Either<XPathException, Sequence> actual = executeQuery(query);
+
+        assertXQResultSimilar(expected, actual);
```

Comment:
@adamretter Why did you choose to check for similarity here?
Does it give a better output in case it does not match or is it a looser check to get around whitespace differences or other minor differences?

Reply:
The difference between `similar and `identical` is referenced from the JavaDoc in `DiffMatcher.java` and documented here https://www.xmlunit.org/api/java/2.6.0/org/xmlunit/builder/DiffBuilder.html#checkForSimilar-- and 

> is it a looser check to get around whitespace differences or other minor differences

Yes.